### DISCUSSION TITLE
Is it possible to save a query that returns empty even though it has bindings?

### DISCUSSION THREAD

**Discussion by nunick** [Budibase/budibase#4298] [Help] [answered]
I have a screen where a user enters a serial number that is used to filter a query through bindings and then auto populates some fields. The query is set up as shown:

![image](https://user-images.githubusercontent.com/94874984/152217379-d7531c0f-94ac-4f97-8793-e7625a78d12f.png)

However, when I run the query, the save button isn't enabled. 

If I add a default value to the binding that queries some data, then the save button will enable. 

![image](https://user-images.githubusercontent.com/94874984/152218114-b0ccabd0-994b-4ac9-ac7a-9c4784dbd698.png)

If the default value is saved, then the page I'd like to automatically query data on (auto populate) will show the data from this test value, which isn't desirable.

A work around I found is if I run the query with the test value, have the save button enabled, remove the test value, I can then save the query.

![image](https://user-images.githubusercontent.com/94874984/152218165-cabe0528-f65e-4747-9d64-12603ee8c743.png)

Is this intended behavior or is there a better convention for saving with a default test value?

**Comment by mjashanks** [comment, marked as answer]
Yea - I understand. I find this to be annoying as well. The reason for this because Budibase needs a schema to use these queries - i.e. it needs to know what fields are returned, so we can bind to them in the Design section. 

When no results are returned from the query, we don't have a schema.

The workaround for this is to manually enter the schema:
![image](https://user-images.githubusercontent.com/3524181/152316114-344ba004-fd10-4cf3-85aa-194aea6a279f.png)

However, it's still probably easier to pull results back, then edit your query / parameters values as needed after. Once the Schema is populated, you can save.
